runtime/common/tests/xcm_tests.rs

```diff
@@ -2156,3 +2157,579 @@ fn less_than_min_xcm_fee_should_not_work() {
+            parachain::Assets::total_supply(relay_asset_id_on_b),
+            fee_amount - min_xcm_fee
+        );
+    });
+}
+
+#[test]
+fn transfer_multicurrencies_should_fail_scenarios() {
+    MockNet::reset();
+
+    let para_a_id = 1;
+    let para_b_id = 2;
+    let para_c_id = 3;
+    let para_a_source_location = create_asset_location(1, para_a_id);
+    let para_b_source_location = create_asset_location(1, para_b_id);
+    let para_c_source_location = create_asset_location(1, para_c_id);
+
+    let para_a_asset_metadata =
+        create_asset_metadata("ParaAToken", "ParaA", 18, 1, None, false, true);
+    let para_b_asset_metadata =
+        create_asset_metadata("ParaBToken", "ParaB", 18, 1, None, false, true);
+    let para_c_asset_metadata =
+        create_asset_metadata("ParaCToken", "ParaC", 18, 1, None, false, true);
+
+    let relay_source_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::parent()));
+    let relay_asset_metadata = create_asset_metadata("Kusama", "KSM", 12, 1, None, false, true);
+    let units_per_sec = 0;
+
+    let a_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &para_a_source_location,
+        &para_a_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let relay_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let b_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &para_b_source_location,
+        &para_b_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let c_asset_id_on_a = register_assets_on_parachain::<ParaA>(
+        &para_c_source_location,
+        &para_c_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    let b_asset_id_on_b = register_assets_on_parachain::<ParaB>(
+        &para_b_source_location,
+        &para_b_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let _ = register_assets_on_parachain::<ParaB>(
+        &para_a_source_location,
+        &para_a_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let relay_asset_id_on_b = register_assets_on_parachain::<ParaB>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    let c_asset_id_on_c = register_assets_on_parachain::<ParaC>(
+        &para_c_source_location,
+        &para_c_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let _ = register_assets_on_parachain::<ParaC>(
+        &para_a_source_location,
+        &para_a_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+    let _ = register_assets_on_parachain::<ParaC>(
+        &relay_source_location,
+        &relay_asset_metadata,
+        Some(units_per_sec),
+        None,
+    );
+
+    let amount = 1000;
+    let weight = 40;
+    assert_ok!(ParaA::execute_with(|| {
```

Comment:
I'm not sure what you're testing here. Could you add a `///` doccomment fn header describing the test scenarios with a few words?

Reply:
First part are supposed to be only must fail cases on the sender side. covering all the error messages.
Added some comments on the asserts.
The second part are must fail case on the receiver part.

Reply:
Please don't make me write comments for all the tests though :D

Reply:
Agree it'd be torture to comment each test case, my idea was to annotate the main fn:
https://github.com/Manta-Network/Manta/pull/576/files/17def4e433d575d75d82e3de7150e0871ce4ae3d#diff-c55c7706ced753270f03a35c10fdb301207231f2224a8e3f3951c5970f546643R2365

With a comment describing the structure you just put in the comment here.
So that when i'm reading the code I know what to expect instead of having to rhyme things together line by line

Reply:
Ok, added the comment. Let me know if it doesn't make sense.